src/nrtk_explorer/app/transforms.py

```diff
@@ -146,11 +146,9 @@ def on_apply_transform(self, *args, **kwargs):
         if len(transformed_image_ids) == 0:
             return
 
-        dataset = get_dataset(self.state.current_dataset)
-
         # Erase current annotations
         dataset_ids = [image_id_to_dataset_id(id) for id in self.state.source_image_ids]
-        for ann in dataset["annotations"]:
+        for ann in self.context.dataset.anns.values():
```

Comment:
`self.context.dataset.dataset['annotations']` for similar reasons as mentioned elsewhere.